backend/src/cache/storage/execution_cache_store.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"strconv"
+
+	model "github.com/kubeflow/pipelines/backend/src/cache/model"
+	"github.com/kubeflow/pipelines/backend/src/common/util"
+)
+
+type ExecutionCacheStoreInterface interface {
+	GetExecutionCache(executionCacheKey string) (*model.ExecutionCache, error)
+	CreateExecutionCache(*model.ExecutionCache) (*model.ExecutionCache, error)
+	DeleteExecutionCache(executionCacheKey string) error
+}
+
+type ExecutionCacheStore struct {
+	db   *DB
+	time util.TimeInterface
+}
+
+func (s *ExecutionCacheStore) GetExecutionCache(executionCacheKey string) (*model.ExecutionCache, error) {
+	r, err := s.db.Table("execution_caches").Where("ExecutionCacheKey = ?", executionCacheKey).Rows()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get execution cache: %q", executionCacheKey)
+	}
+	defer r.Close()
+	executionCaches, err := s.scanRows(r)
```

Comment:
Is there any reason not to use  `.Find` or `.Scan`? `s.db.Table("execution_caches").Where("ExecutionCacheKey = ?", executionCacheKey).Scan(&cacheEntries)`. http://gorm.io/docs/query.html

Reply:
Here is to have more information about what is happening inside of Scan().